### ISSUE TITLE
Accessing protected computed property does not produce a compiler error

### ISSUE THREAD

**Issue by parzhitsky** [microsoft/TypeScript#60974]
### 🔎 Search Terms

protected computed no error bug

### 🕗 Version & Regression Information

This is the behavior in every version I tried (v3.3.3333, v4.0.5, v5.7.3), and I reviewed the FAQ for entries about **Common "Bugs" That Aren't Bugs**

### ⏯ Playground Link

https://tsplay.dev/wjdn7m

### 💻 Code

```ts
declare class Foo {
  protected readonly baz = 42
}

declare class Bar {
  readonly foo: Foo
}

declare const bar: Bar

// @ts-expect-error
console.log(bar.foo["baz"].toFixed(2))
```

### 🙁 Actual behavior

- (without `// @ts-expect-error`): No errors
- (with `// @ts-expect-error`): Compiler error `Unused '@ts-expect-error' directive`

### 🙂 Expected behavior

- (without `// @ts-expect-error`): Compiler error `Property […] is protected and only accessible within class 'Foo' and its subclasses`
- (with `// @ts-expect-error`): No errors

**Comment by jcalz** [comment]
This is working as intended, duplicate #19335

**Comment by parzhitsky** [comment]
_(I'd respond in the original issue, but it is limited to collaborators only)_

I think, there needs to be a distinction between intended usage of bracket notation where dot notation is available, and unavoidable usage of bracket notation where dot notation is not available. I.e., my original use case is using a symbol as property key, – I can't _not_ use bracket notation:

```ts
const key = Symbol('key')

declare class Foo {
  protected readonly [key]: number
}

declare class Bar {
  readonly foo: Foo
}

declare const bar: Bar

console.log(bar.foo[key].toFixed(2)) // no errors 🙁
```

_[Try it](https://tsplay.dev/weYpgw)._

**Comment by parzhitsky** [comment]
Should I open a feature request?

**Comment by jcalz** [comment]
You could possibly do file a feature request, but similar things like #50485 have been declined. Not sure if someone has brought up symbol keys before, but I imagine you'll get the same response.

**Comment by RyanCavanaugh** [comment]
Enough people dislike #19335 that there's probably room for something like `strictClassPropertyAccess` that would disallow the private-evading syntax carve-out. I'm sure a lot of codebases don't actually use it intentionally.

**Comment by parzhitsky** [comment]
@MartinJohns I believe this mechanism would be similar to (if not the same as) one that works [here](https://tsplay.dev/mqMkrm):

```ts
("value") as const // no error
("va" + "lu" + "e") as const // error
```
